src/Opdex.Platform.Infrastructure/Data/Handlers/Indexer/PersistIndexerUnlockCommandHandler.cs

```diff
@@ -21,18 +22,27 @@ public class PersistIndexerUnlockCommandHandler : IRequestHandler<PersistIndexer
             WHERE {nameof(IndexLockEntity.InstanceId)} = @{nameof(IndexLockEntity.InstanceId)};";
 
     private readonly IDbContext _context;
+    private readonly ILogger<PersistIndexerUnlockCommandHandler> _logger;
     private readonly string _instanceId;
 
-    public PersistIndexerUnlockCommandHandler(IDbContext context, OpdexConfiguration opdexConfiguration)
+    public PersistIndexerUnlockCommandHandler(IDbContext context, ILogger<PersistIndexerUnlockCommandHandler> logger, OpdexConfiguration opdexConfiguration)
     {
         _context = context ?? throw new ArgumentNullException(nameof(context));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         _instanceId = opdexConfiguration?.InstanceId ?? throw new ArgumentNullException(nameof(context));
     }
 
-    public async Task<bool> Handle(PersistIndexerUnlockCommand request, CancellationToken cancellationToken)
+    protected override async Task Handle(PersistIndexerUnlockCommand request, CancellationToken cancellationToken)
     {
-        var command = DatabaseQuery.Create(SqlQuery, new { InstanceId = _instanceId }, CancellationToken.None);
-        var result = await _context.ExecuteCommandAsync(command);
-        return result == 1;
+        try
+        {
+            var command = DatabaseQuery.Create(SqlQuery, new { InstanceId = _instanceId }, CancellationToken.None);
+            var result = await _context.ExecuteCommandAsync(command);
+            if (result == 0) throw new NoRowsAffectedException("Indexer appears to have not unlocked.");
+        }
+        catch (Exception ex)
+        {
+            _logger.LogCritical(ex, "Failed to unlock indexer.");
+        }
```

Comment:
You may have a reason for throwing and immediately catching, but it looks to me that you could just log here instead of throwing, catching, then logging.

Reply:
The intent was to follow the same log message and be able to see the cause of failure in the form of the exception. Thinking about it again, I agree the exception is not necessary.